controllers/core/flagd/resources/deployment.go

```diff
@@ -100,9 +109,14 @@ func (r *FlagdDeployment) GetResource(ctx context.Context, flagd *api.Flagd) (cl
 		return nil, errors.New("no flagd container has been injected into deployment")
 	}
 
-	deployment.Spec.Template.Spec.ImagePullSecrets = imagePullSecrets
-
 	// override settings for the injected container for flagd standalone deployment mode
+	deployment.Spec.Template.Spec.ImagePullSecrets = imagePullSecrets
+	if len(r.FlagdConfig.Labels) > 0 {
+		maps.Copy(deployment.Spec.Template.ObjectMeta.Labels, r.FlagdConfig.Labels)
+	}
+	if len(r.FlagdConfig.Annotations) > 0 {
+		maps.Copy(deployment.Spec.Template.ObjectMeta.Annotations, r.FlagdConfig.Annotations)
+	}
```

Comment:
Are these lines actually necessary?

I think this is taken care of now by this: https://github.com/open-feature/open-feature-operator/pull/681/files#diff-ea25af66f2087ae4e88440e1dcf34877524a2689948abd770c29116cdaa8c3b1R80-R81

Reply:
Update: I removed these lines. I think they are redundant. See net changes: https://github.com/open-feature/open-feature-operator/pull/681/commits/59686ced0dfb820fbf5f49b9d28daa65e424eed7